SCloudRandomizer/MainViewController.m

```diff
@@ -65,7 +67,17 @@ - (void)viewWillAppear:(BOOL)animated
         
         if (self.mySearchParams.hasParamsChanged)
         {
-            [self getTracks];
+            
+            if ([self.player isPlaying])
+            {
+                [self refreshTrackList];
+                [self doPlayNextSong];
```

Comment:
soPlayNextSong already seems to be calling refereshTrackList so we probably don't need it here?